Buildings/Fluid/MixingVolumes/MixingVolumeEvaporation.mo

```diff
@@ -26,53 +28,157 @@ model MixingVolumeEvaporation
         rotation=0,
         origin={110,70})));
 
+  // Parameter
+  parameter Modelica.SIunits.Volume V "Total volume";
+
+// Variables
+  MediumWat.ThermodynamicState state_saturated
+    "Saturated state";
+  MediumSte.AbsolutePressure p(
+    start=p_start,
+    stateSelect=if massDynamics == Modelica.Fluid.Types.Dynamics.SteadyState
+    then StateSelect.default else StateSelect.prefer)
+    "Pressure inside volume";
+  MediumSte.Temperature T(start=T_start) "Temperature inside volume";
+  Modelica.SIunits.Volume VSte(final start=VSte_start) "Volume of steam vapor";
+  Modelica.SIunits.Volume VWat(
+    start=VWat_start,
+    stateSelect=if massDynamics == Modelica.Fluid.Types.Dynamics.SteadyState
+    then StateSelect.default else StateSelect.prefer)
+    "Volume of liquid water phase";
+  MediumSte.SpecificEnthalpy hSte "Specific enthalpy of steam vapor";
+  MediumWat.SpecificEnthalpy hWat "Specific enthalpy of liquid water";
+  MediumSte.Density rhoSte "Density of steam vapor";
+  MediumWat.Density rhoWat "Density of liquid water";
+  Modelica.SIunits.Mass m "Total mass of volume";
+  Modelica.SIunits.Energy U(final start=U_start) "Internal energy";
+  Modelica.SIunits.MassFlowRate mWat_flow "Feed water mass flow rate";
+  Modelica.SIunits.MassFlowRate mSte_flow "Steam mass flow rate";
+
+  // Initialization
+  parameter Modelica.SIunits.Volume VWat_start=V/2
+    "Start value of liquid volume"
+    annotation (Dialog(tab="Initialization"));
+  final parameter Modelica.SIunits.Volume VSte_start= V-VWat_start
+  "Start value of steam vapor volume";
+  final parameter MediumWat.ThermodynamicState state_start = MediumWat.setState_pTX(
+      T=T_start,
+      p=p_start,
+      X=MediumWat.X_default) "Medium state at default values";
+  final parameter Modelica.SIunits.Density rhoWat_start=MediumWat.density(
+    state=state_start) "Density, used to compute fluid mass";
+  final parameter Modelica.SIunits.Density rhoSte_start=MediumSte.density(
+    state=state_start) "Density, used to compute fluid mass";
+  final parameter Modelica.SIunits.SpecificEnthalpy hSteStart=
+    MediumSte.specificEnthalpy_pTX(p_start, T_start, MediumSte.X_default)
+    "Start value for specific enthalpy";
+  final parameter Modelica.SIunits.Energy U_start=
+    VWat_start*rhoWat_start*MediumWat.specificEnthalpy(state_start) +
+    VSte_start*rhoSte_start*MediumSte.specificEnthalpy(state_start) -
+    p_start*V
+    "Starting internal energy";
```

Comment:
So, I am a bit confused, because at face value, I would say that these statements are not needed, since the calculation for `U_start` is the same as the calculation for `U` and Modelica should be able to propagate the general equation to determine the start value. But I stand corrected because I tried to it on my end, and Modelica does complain that U is missing a start value.

Anyway. You don't need to change anything here, but I just wanted to verbalize my confusion.

Reply:
Thanks for your thoughts! It's helpful for my understanding as well.